pkg/reconciler/revision/resources/deploy.go

```diff
@@ -139,20 +139,29 @@ func certVolume(secret string) corev1.Volume {
 	}
 }
 
-func rewriteUserProbe(p *corev1.Probe, userPort int) {
+func addLivenessProbeHeader(p *corev1.Probe) {
 	if p == nil {
 		return
 	}
 	switch {
 	case p.HTTPGet != nil:
-		p.HTTPGet.Port = intstr.FromInt(userPort)
 		// With mTLS enabled, Istio rewrites probes, but doesn't spoof the kubelet
```

Comment:
diff is a bit suboptimal. I extracted the header setting as we need it in UC + sidecar and have `rewriteUserLivenessProbe` just for the port rewrite on the UC